File: cpp/assert/assert_array.cpp
// Make sure function argument is array

void assert_array(const v8::FunctionCallbackInfo < v8::Value > &aArguments, v8::Local<v8::Value> aValue, const char* aVariableName, const char* aFunctionName) {
    // Make sure function argument is array
    if (!aValue->IsArray()) {
        hint_args(aFunctionName, aArguments);
        fatal_msg_stack(globalIsolate, EXIT_ARGS, "variable '%s' in function '%s' should be array", aVariableName, aFunctionName);
        exit(EXIT_UNREACHABLE);
    }
}

File: js/assert/assert_array.js
// Assert that value is an array
// linter: ngspicejs-lint --internal
"use strict";

function assert_array(aValue, aVariableName, aFunctionName) {
    // Assert that value is an array
    assert_arguments_length(arguments, 3, 3, 'assert_array(array,variable_name,function_name)');
    if (!Array.isArray(aValue)) {
        throw new Exception(aVariableName + ' in function ' + aFunctionName + ' should be array but is ' + typeof aValue);
    }
}

globalThis.exports = {assert_array};
Internal.assert_array = assert_array;
